dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/VisualizationController.java

```diff
@@ -141,6 +149,38 @@ public void postProcessResponseEntity( Visualization visualization, WebOptions o
                     period.setName( i18nFormat.formatPeriod( period ) );
                 }
             }
+
+            setDataDimensionItems( visualization );
         }
     }
+
+    private void setDataDimensionItems( Visualization visualization )
+    {
+        List<BaseDimensionalItemObject> dimensionalItemObjectList = visualization.getDataDimensionItems()
+            .stream()
+            .filter( ddi -> ddi.getExpressionDimensionItem() != null )
+            .flatMap( ddi -> ExpressionDimensionItemHelper.getExpressionItems( manager, ddi ).stream() )
+            .collect( Collectors.toList() );
+
+        visualization.getDataDimensionItems().addAll( dimensionalItemObjectList.stream().map( d -> {
+            DataDimensionItem ddi = new DataDimensionItem();
+
+            switch ( d.getDimensionItemType() )
```

Comment:
I am trying to understand what this code does. Why do handle DataElement/Indicator/ProgramDataElementDimensionItem but not DataElementOperand, ReportingRate, ProgramIndicator, etc?